packages/x-date-pickers/src/locales/arEg.ts

```diff
@@ -0,0 +1,97 @@
+import { PickersLocaleText } from './utils/pickersLocaleTextApi';
+import { getPickersLocalization } from './utils/getPickersLocalization';
+
+// This object is not Partial<PickersLocaleText> because it is the default values
+
+const arEGPickers: PickersLocaleText<any> = {
+  // Calendar navigation
+  previousMonth: 'الشهر السابق',
+  nextMonth: 'الشهر التالي',
+
+  // View navigation
+  openPreviousView: 'افتح الشاشة السابقة',
+  openNextView: 'افتح الشاشة التالية',
+  calendarViewSwitchingButtonAriaLabel: (view) =>
+    view === 'year'
+      ? 'شاشة السنوات مفتوحة، قم بالتبديل إلى شاشة التقويم'
+      : 'شاشة التقويم مفتوحة، قم بالتبديل إلى شاشة  السنوات',
+
+  // DateRange labels
+  start: 'بداية',
+  end: 'نهاية',
+  startDate: 'بداية التاريخ',
+  startTime: 'بداية التوقيت',
+  endDate: 'نهاية التاريخ',
+  endTime: 'نهاية التوقيت',
+
+  // Action bar
+  cancelButtonLabel: 'إلغاء',
+  clearButtonLabel: 'مسح',
+  okButtonLabel: 'موافق',
+  todayButtonLabel: 'اليوم',
+
+  // Toolbar titles
+ datePickerToolbarTitle: 'اختر التاريخ',
+  dateTimePickerToolbarTitle: 'اختر التاريخ والوقت',
+  timePickerToolbarTitle: 'اختر الوقت',
+  dateRangePickerToolbarTitle: 'اختر نطاق التاريخ',
+
+  // Clock labels
+  clockLabelText: (view, time, adapter) =>
+    `اختر ${view}. ${
```

Comment:
The `view` variable here will be in English.
Could you please check any non-English locale for a correct translation?
You can check `frFR` or any other locale for an example of how it is handled. 😉